Transport/Simple/Transport.H

```diff
@@ -11,427 +11,778 @@
+    amrex::The_Arena()->free(trans_parm.trans_sig);
+    amrex::The_Arena()->free(trans_parm.trans_dip);
+    amrex::The_Arena()->free(trans_parm.trans_pol);
+    amrex::The_Arena()->free(trans_parm.trans_zrot);
+    amrex::The_Arena()->free(trans_parm.trans_fitmu);
+    amrex::The_Arena()->free(trans_parm.trans_fitlam);
+    amrex::The_Arena()->free(trans_parm.trans_fitdbin);
+    amrex::The_Arena()->free(trans_parm.trans_nlin);
   }
+};
 
-  MW_m *= MW_m;
-  amrex::Real sigma_M = std::cbrt(sigma_M_3);
-  amrex::Real InvSigma1 = 1.0 / sigma_M;
-  amrex::Real InvSigma3 = 1.0 / sigma_M_3;
-  Epsilon_M *= InvSigma3;
-  amrex::Real Tstar = Temp / Epsilon_M;
-  amrex::Real Tcm = 1.2593* Epsilon_M;
-  amrex::Real Vcm = 1.8887 * sigma_M_3;
-  Omega_M *= InvSigma3;
-  MW_m = MW_m * InvSigma3 * InvSigma1 / (Epsilon_M * Epsilon_M);
-  DP_m_4 = DP_m_4 * sigma_M_3 * Epsilon_M;
-  amrex::Real DP_red_4 =  297.2069113e6 * DP_m_4 / (Vcm * Vcm * Tcm * Tcm);
-  amrex::Real y = Vcm * rholoc / (6.0 * wbar);
-  amrex::Real A[10];
-  for (int i = 0; i<10; ++i) {
-    A[i] = trans_parm->Afac[4*i] + trans_parm->Afac[4*i+1] * Omega_M + trans_parm->Afac[4*i+2] * DP_red_4 + trans_parm->Afac[4*i+3]*KappaM;
+// FIXME remove ifdef when compiling in all the EOS
+#ifdef PELEC_USE_SRK
+template <>
+struct CloseTransport<eos::SRK>
+{
+  void operator()()
+  {
+    amrex::The_Device_Arena()->free(trans_parm_g);
+
+    amrex::The_Arena()->free(trans_parm.trans_wt);
+    amrex::The_Arena()->free(trans_parm.trans_iwt);
+    amrex::The_Arena()->free(trans_parm.trans_eps);
+    amrex::The_Arena()->free(trans_parm.trans_sig);
+    amrex::The_Arena()->free(trans_parm.trans_dip);
+    amrex::The_Arena()->free(trans_parm.trans_pol);
+    amrex::The_Arena()->free(trans_parm.trans_zrot);
+    amrex::The_Arena()->free(trans_parm.trans_fitmu);
+    amrex::The_Arena()->free(trans_parm.trans_fitlam);
+    amrex::The_Arena()->free(trans_parm.trans_fitdbin);
+    amrex::The_Arena()->free(trans_parm.trans_nlin);
+
+    amrex::The_Arena()->free(trans_parm.Afac);
+    amrex::The_Arena()->free(trans_parm.Bfac);
+    amrex::The_Arena()->free(trans_parm.Kappai);
+    amrex::The_Arena()->free(trans_parm.Sigmaij);
+    amrex::The_Arena()->free(trans_parm.omega);
   }
-  amrex::Real G1 = (1.0 - 0.5*y)/((1.0-y)*(1.0-y)*(1.0-y));
-  amrex::Real G2 = ( A[0]*(1.0 - std::exp(-A[3]*y))/y + A[1] * G1 * std::exp(A[4] * y)
-	+ A[2]*G1 ) / (A[0]*A[3] + A[1] + A[2]);
-  amrex::Real eta_P = (36.344e-6*std::sqrt(MW_m*Tcm)/std::pow(Vcm, 2.0/3.0) )
-    *A[6]*y*y*G2*exp(A[7] + A[8]/Tstar + A[9]/(Tstar*Tstar));
-
-  // Set nonideal viscosity
-  if (wtr_get_mu) {
-    mu = mu * (1.0/G2 + A[5] * y ) + eta_P;
+};
+#endif
+
+template <typename EOSType>
+struct SRKChungCorrections
+{
+  template <class... Args>
+  AMREX_GPU_HOST_DEVICE
+  AMREX_FORCE_INLINE
+  void operator()(Args...)
+  {
   }
+};
+
+// FIXME remove ifdef when compiling in all the EOS
+#ifdef PELEC_USE_SRK
+template <>
+struct SRKChungCorrections<eos::SRK>
```

Comment:
IMO, this should be called NonidealChungCorrections because the Chung corrections for transport aren't specific to the nonideal EOS

Reply:
done